### PR TITLE
Data Infra for Training

### PR DESCRIPTION
**Pull request by bgenchel**
started a new branch because there was some mismatch with the previous branch and remote, so this seemed easier. 

Not infra as in backend, obv. but code structure and tests for datasets, with guitarset implemented as a first inclusion / example.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -19,6 +19,7 @@ classifiers = [
 ]
 dependencies = [
 	"coremltools; platform_system == 'Darwin'",
+	"apache_beam",
```
Comment (drubinstein) [inline pyproject.toml:22]: This shouldn't be in the general project deps. This should be in extras only.

File: basic_pitch/data/datasets/guitarset.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+    pipeline_options = {
+        "runner": known_args.runner,
+        "job_name": f"guitarset-tfrecords-{time_created}",
+        "machine_type": "e2-standard-4",
+        "num_workers": 25,
+        "disk_size_gb": 128,
+        "experiments": ["use_runner_v2"],
+        "save_main_session": True,
+        "worker_harness_container_image": known_args.worker_harness_container_image,
```
[the remark this answers is not in the document]
Comment (bgenchel) [reply basic_pitch/data/datasets/guitarset.py:166]: let's discuss in meeting

File: basic_pitch/note_creation.py [hunk 3]
```diff
@@ -110,7 +110,11 @@ def model_output_to_notes(
     )
 
 
-def sonify_midi(midi: pretty_midi.PrettyMIDI, save_path: Union[pathlib.Path, str], sr: Optional[int] = 44100) -> None:
+def sonify_midi(
```
[the remark this answers is not in the document]
Comment (bgenchel) [reply basic_pitch/note_creation.py:113]: I don't think so, but I did run the formatter via tox. I can switch it all back.

[the remark this answers is not in the document]
Comment (drubinstein) [reply basic_pitch/note_creation.py:113]: In the new ones at least

[the remark this answers is not in the document]
Comment (bgenchel) [reply basic_pitch/note_creation.py:113]: I switched it all back, but after running tox it does seem that black wants to reformat two long lines into multi line blocks even though they don't exceed the line limit. Wonder if it's some other rule

File: pyproject.toml [hunk 4]
```diff
@@ -49,9 +49,18 @@ namespaces = false
 
 [project.scripts]
 basic-pitch = "basic_pitch.predict:main"
+bp-download = "basic_pitch.data.download:main"
 
 [project.optional-dependencies]
+train = [
```
Comment (drubinstein) [inline pyproject.toml:55]: I think naming this `data` would be more apt.

**Review by drubinstein — APPROVED** [review]
2 small nits but I think this is overall ready to merge!

File: pyproject.toml [hunk 5]
```diff
@@ -49,9 +49,18 @@ namespaces = false
...
+	"basic_pitch[tf]",
```
[the remark this answers is not in the document]
Comment (bgenchel) [reply pyproject.toml:56]: what is the issue?

[the remark this answers is not in the document]
Comment (drubinstein) [reply pyproject.toml]: Seems fixed now on github. Weird.

File: tox.ini [hunk 6]
```diff
@@ -27,6 +27,11 @@ setenv =
     TEST_SOURCE = {toxinidir}/tests
     LINE_LENGTH = "120"
 
+[testenv:train]
+deps = -e .[train]
```
Comment (drubinstein) [inline tox.ini:31]: I believe this has to be data to now that train is now data

Comment (drubinstein) [inline tox.ini:31]: I believe you also need to add `train` to the envlist at the top of the file.

**Review by drubinstein — APPROVED** [review]

**Pull request merged**